static/extensions/MubiLop/numutils.js

```diff
@@ -0,0 +1,377 @@
+                defaultValue: 'NaN'
+              }
+            }
+          },
+          {
+            opcode: 'parseNumber',
+            blockType: BlockType.REPORTER,
+            text: 'parse number from text [TEXT]',
+            arguments: {
+              TEXT: {
+                type: ArgumentType.STRING,
+                defaultValue: '123.45'
+              }
+            }
+          },
+          {
+            opcode: 'clamp',
+            blockType: BlockType.REPORTER,
+            text: 'clamp [NUMBER] between [MIN] and [MAX]',
+            arguments: {
+              NUMBER: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 50
+              },
+              MIN: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 0
+              },
+              MAX: {
+                type: ArgumentType.NUMBER,
+                defaultValue: 100
+              }
+            }
+          }
+        ]
+      };
+    }
+
+    toLocaleString(args) {
+      try {
+        const locale = args.LOCALE === 'system' ? this.systemLocale : args.LOCALE;
+        return Number(args.NUMBER).toLocaleString(locale);
+      } catch (e) {
+        throw new Error('Error: Invalid number or locale');
+      }
+    }
+
+    toFixed(args) {
+      try {
+        return Number(args.NUMBER).toFixed(args.DECIMALS);
+      } catch (e) {
+        throw new Error('Error: Invalid input');
+      }
+    }
+
+    toPrecision(args) {
+      try {
+        return Number(args.NUMBER).toPrecision(args.PRECISION);
+      } catch (e) {
+        throw new Error('Error: Invalid input');
+      }
+    }
+
+    toExponential(args) {
+      try {
+        return Number(args.NUMBER).toExponential(args.DECIMALS);
+      } catch (e) {
+        throw new Error('Error: Invalid input');
+      }
+    }
+
+    formatCurrency(args) {
+      try {
+        return Number(args.NUMBER).toLocaleString('en-US', {
+          style: 'currency',
+          currency: args.CURRENCY
+        });
+      } catch (e) {
+        throw new Error('Error: Invalid input');
+      }
+    }
+
+    formatPercentage(args) {
+      try {
+        const number = Number(args.NUMBER);
+        return `${(number * 100).toFixed(args.DECIMALS)}%`;
+      } catch (e) {
+        throw new Error('Error: Invalid input');
+      }
+    }
+
+    roundNumber(args) {
+      try {
+        const multiple = Number(args.MULTIPLE);
+        return Math.round(args.NUMBER / multiple) * multiple;
+      } catch (e) {
+        throw new Error('Error: Invalid input');
+      }
+    }
+
+    isInteger(args) {
+      return Number.isInteger(Number(args.NUMBER));
+    }
+
+    isFinite(args) {
+      return Number.isFinite(Number(args.NUMBER));
+    }
+
+    isNaN(args) {
+      return Number.isNaN(Number(args.NUMBER));
+    }
+
+    parseNumber(args) {
+      try {
+        const parsed = parseFloat(args.TEXT);
+        return Number.isNaN(parsed) ? 'Error: Invalid number' : parsed;
```

Comment:
if its NaN just return NaN no big deal you have a block to check if it is just let people use that if they dont want NaN